veracruz-utils/src/csr.rs

```diff
@@ -227,7 +227,8 @@ pub fn generate_csr(
     template: &CsrTemplate,
     private_key: &EcdsaKeyPair,
 ) -> Result<Vec<u8>, CertError> {
-    let public_key = private_key.public_key().as_ref().clone();
+    #[allow(clippy::clone_double_ref)]
```

Comment:
Hmm, this seems like it should be easy to fix, no?  How about something like:

```rust
let public_key = (*private_key.public_key().as_ref()).clone();
```

Does that work?  If not, then we need to come up with a fix as I don't think we should allow that Clippy lint...

Reply:
@dominic-mulligan-arm yeah I allowed it only temporarily, I was hesitant to fix it because I am not fully understanding the context of the code. I tried your solution but it doesn't work. 

I came up with this before 
```
<&[u8]>::clone(&private_key.public_key().as_ref())
```
what do you think of this ?

Reply:
What's wrong with just `let public_key =  private_key.public_key().as_ref();`?

(On my branch I've deleted the line anyway so I can't easily test ...)

Reply:
Yes, but what's wrong with just `let public_key = private_key.public_key().as_ref();`, without a `clone`?

Reply:
Yeah you are right. I don't understand what is the reason behind `clone`. I would prefer to remove it if this is Ok

Reply:
Yes, I think the `clone` is unnecessary. It's ironic that neither Clippy nor I immediately noticed that. Clippy spouted some overcomplicated stuff about cloning a double-reference, and I complained about the extra space. Neither of us noticed the main thing!

Reply:
:sweat_smile: 
I removed it and finished your previous requested changes

Reply:
Great, thanks for addressing that @Mo-Fatah!